### PR TITLE
Remove copy-pasting of example

### PR DESCRIPTION
**Pull request by lukaszstolarczuk**
- remove copy-pasted example from README and replace with link to proper directory
- update examples to be aligned with manpage content & latest API
- make examples included dynamically into manpage

---
This change is [Reviewable](https://reviewable.io/reviews/pmem/pmemkv/370)

### CODE REVIEW

File: doc/CMakeLists.txt [hunk 1]
```diff
@@ -32,16 +32,26 @@
 add_check_whitespace(man
 	${CMAKE_CURRENT_SOURCE_DIR}/*.md)
 
+# prepare C_EXAMPLE content to inject, based on example's code;
+# remove comments (with license and file description) and trailing spaces
```
Comment (marcinslusarz) [inline doc/CMakeLists.txt]: why not fix trailing spaces?

**Comment by lukaszstolarczuk** [comment]
I've updated files with fixes to all Marcin's issues

File: doc/CMakeLists.txt (hunk 1 above)
Comment (lukaszstolarczuk) [reply doc/CMakeLists.txt]: Done.

File: examples/README [hunk 2]
```diff
@@ -1,2 +1,9 @@
-This directory contains examples for pmemkv, the library providing
-key-value datastore optimized for persistent memory.
+This directory contains c and c++ examples for pmemkv, the library
```
[the remark this answers is not in the document]
Comment (lukaszstolarczuk) [reply examples/README]: Done.

File: examples/pmemkv_basic_cpp/pmemkv_basic.cpp [hunk 3]
```diff
@@ -64,7 +63,7 @@ int main(int argc, char *argv[])
 	ret = pmemkv_config_put_uint64(cfg, "force_create", 1);
 	assert(ret == PMEMKV_STATUS_OK);
 
-	LOG("Starting engine");
+	LOG("Opening pmemkv databse with 'cmap' engine");
```
[the remark this answers is not in the document]
Comment (lukaszstolarczuk) [reply examples/pmemkv_basic_cpp/pmemkv_basic.cpp]: Done.

File: examples/pmemkv_basic_cpp/pmemkv_basic.cpp [hunk 4]
```diff
@@ -53,7 +52,7 @@ int main(int argc, char *argv[])
 		exit(1);
 	}
 
-	LOG("Creating config");
+	LOG("Creating config, see libpmemkv_config(3) for more detailed example");
```
[the remark this answers is not in the document]
Comment (lukaszstolarczuk) [reply examples/pmemkv_basic_cpp/pmemkv_basic.cpp]: Done.

File: examples/pmemkv_basic_c/pmemkv_basic.c [hunk 5]
```diff
@@ -72,7 +71,7 @@ int main(int argc, char *argv[])
 	s = pmemkv_config_put_uint64(cfg, "force_create", 1);
 	assert(s == PMEMKV_STATUS_OK);
 
-	LOG("Starting engine");
+	LOG("Opening pmemkv databse with 'cmap' engine");
```
[the remark this answers is not in the document]
Comment (lukaszstolarczuk) [reply examples/pmemkv_basic_c/pmemkv_basic.c]: Done.

File: examples/pmemkv_basic_c/pmemkv_basic.c [hunk 6]
```diff
@@ -61,7 +60,7 @@ int main(int argc, char *argv[])
 		exit(1);
 	}
 
-	LOG("Creating config");
+	LOG("Creating config, see libpmemkv_config(3) for more detailed example");
```
[the remark this answers is not in the document]
Comment (lukaszstolarczuk) [reply examples/pmemkv_basic_c/pmemkv_basic.c]: Done.

File: examples/README [hunk 7]
```diff
@@ -1,2 +1,9 @@
...
+providing key-value datastore optimized for persistent memory.
+For more information see libpmemkv(3).
+
+pmemkv_basic_c/pmemkv_basic.c -- contains basic example workflow
+		of c application
+
+pmemkv_basic_cpp/pmemkv_basic.cpp -- contains basic example workflow
+		of c++ application
```
[the remark this answers is not in the document]
Comment (lukaszstolarczuk) [reply examples/README]: Done.

File: examples/README [hunk 8]
```diff
@@ -1,2 +1,9 @@
-This directory contains examples for pmemkv, the library providing
-key-value datastore optimized for persistent memory.
+This directory contains c and c++ examples for pmemkv, the library
+providing key-value datastore optimized for persistent memory.
+For more information see libpmemkv(3).
+
+pmemkv_basic_c/pmemkv_basic.c -- contains basic example workflow
+		of c application
```
[the remark this answers is not in the document]
Comment (lukaszstolarczuk) [reply examples/README]: Done.

**Pull request merged**